hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -1350,7 +1355,7 @@ def main(argv=sys.argv, json_writer=simple_json_writer):  # pragma: no cover
             provider.debug_completed_nodes()
         
         # best effort cleanup.
-        provider.periodic_cleanup()
+        provider.periodic_cleanup(skip_templates=(cmd == "templates"))
```

Comment:
We should add a comment here about the genesis of skip_templates and how it should always be true when the HF calls "templates" (i.e. to avoid an infinite loop / internal deadlock in HF).